### DISCUSSION TITLE
Why a Bootcamp Was Right for Me ⭐

### DISCUSSION THREAD

**Discussion by mnwana** [community/community#56170] [Programming Help]
**Thinking about taking a Full Stack Bootcamp 👢? Here are some of the reasons that I had, that helped me decide–maybe they can help you too! ⬇️**

Prior to my bootcamp experience, I had been exposed to coding in Python for almost a decade, through my university and work experience as a Data Analyst. I always enjoyed the building part of the work and wanted to build more 🛠️. I could create a data pipeline or automate a task, but there was no way I could create my own end-to-end app or contribute to a codebase. I tried to do some self-learning but often felt lost with all of the new concepts. After some research and internal back and forth, I decided a bootcamp was right for me. 

If you’re on GitHub, you most likely fall into one of three buckets; (1) someone with programming experience, (2) a GitHub employee, or (3) someone looking to learn more about programming. If you’re in that last bucket, keep reading to hear my experiences with a Full Stack Bootcamp through [Columbia University](https://bootcamp.cvn.columbia.edu/coding/) and get a sense if a bootcamp is right for you 👍. 

### Why I Chose the Bootcamp Route

As mentioned, I had previous experience in Python. You can’t get a degree in Data Analysis or work as a true Data Analyst without knowing some Python 🐍, but it’s typically going to be focused on data analysis, engineering, and science use cases such as statistical modeling or data pipelines.

One day I had an idea for a website (or app) that I thought would be great to try to build. It felt like it would fit a niche need and not be too hard to build. I was wrong.  Not knowing where to start, I did some Googling and YouTubing and found some tutorials on Django, and hoped my Python knowledge would provide me with a  good enough base. I was immediately lost and didn’t know what anything meant, given I barely knew the basics of HTML. Self-learning was not going to be enough to get this off the ground. 

A few weeks passed and I hadn’t made a dent in the self-learning books I’d purchased and hadn’t written a line of code but I really wanted to, I just felt hopeless. I saw bootcamp ads everywhere for the Flatiron School, Columbia University, and various other options and decided I would apply and take a step in that direction.

### How I Chose My Bootcamp
Choosing a bootcamp wasn’t all that straightforward but it eventually came down to a handful of criteria. I paid close attention to the class schedules, duration of the program, cost 💰, and the syllabus. 

Most important thing for me was the bootcamp cost. If I couldn’t afford it, I wasn’t going to do it. Everyone’s budget is different and this was the best option for me. I chose a program that was the average price I’d seen in my research but also offered flexible payment plans to avoid inundating myself with debt. While the cost might be a lot for many, there are some programs that offer discounts to marginalized groups so make sure to do your own research. 

The bootcamp length and schedule were also key factors for me, as I didn’t want to spend an entire year in school. I wanted to quickly get the skills I needed while being able to maintain my full-time job. I found that part-time was typically twice the duration of full-time programs and the same cost, but with price being a key factor, continuous income was important!

Finally, I paid close attention to the syllabus for each bootcamp. I took into account the types of projects I wanted to develop and skills necessary to move forward. I cross-referenced the syllabus with those desires and researched the most common skillsets and languages of software developers. Picking a bootcamp that teaches you outdated or niche tools isn’t great, unless you have very specific outcome goals. Look up the role you might want for your next career and find a bootcamp that will put those required skills on your resume ✅. 

### Bootcamp Experience
Overall, my bootcamp experience was fast-paced. I had some late nights after work trying to debug code 🐞, or simply just working through an assigned project because I was on a roll and didn’t want to stop. Those moments made me realize this was a good investment for me. 

The bootcamp offered great instructors who were available for help, smart, kind, and diverse students to work with, and lessons on the topics I was looking to learn. I’m currently taking some self-learning courses that feel like a walk in the park because of the solid foundation I received in bootcamp. I’m working through my own app development more confidently this time around.

If you’ve been wondering how to make your next move into software development, I’d highly recommend taking a bootcamp to accelerate that move!

**Comment by Mirza-Ahmad** [comment]
Bootcamps offer immersive, hands-on learning experiences that can help you build a strong foundation in AI concepts, tools, and techniques.
These might include help preparing for job interviews, resume reviews, career coaching, and salary negotiation training. Networking. Machine learning bootcamps are a great place to connect with professionals in the field, from expert instructors to other students.

**Reply by isaiah777-sys** [reply]
> Bootcamps offer immersive, hands-on learning experiences that can help you build a strong foundation in AI concepts, tools, and techniques. These might include help preparing for job interviews, resume reviews, career coaching, and salary negotiation training. Networking. Machine learning bootcamps are a great place to connect with professionals in the field, from expert instructors to other students.

Awesome, you nailed it!

**Comment by jyotiprakash111** [comment]
When I was considering ways to acquire new skills and advance in my professional journey, I carefully evaluated various options, including traditional academic routes and self-study. After careful consideration, I ultimately decided that a bootcamp offered the most practical and efficient path towards achieving my goals.

**Reply by isaiah777-sys** [reply]
> When I was considering ways to acquire new skills and advance in my professional journey, I carefully evaluated various options, including traditional academic routes and self-study. After careful consideration, I ultimately decided that a bootcamp offered the most practical and efficient path towards achieving my goals.

That's true. Most most things you learn in school are not hands on and do not necessary involve you solving problems

**Comment by dockter34** [comment]
It's been a long 24/7contribuition from my part with no help and less info do to so many projects created by dumb people that have no light but even less faith so I didn't fight the fight just to quit when everything is coming to an end, I will start my boot camp.🧐🧐🧐🤔😇

**Reply by Sagir777** [reply]
How I can do it know

**Comment by Radman0915** [comment]
#

**Comment by abwlfdl** [comment]
Hi.thanks 🌹💗🥺🙈

**Reply by abwlfdl** [reply]
Good morning 🌄
OK 🆗

**Comment by gpnarks** [comment]
> **Thinking about taking a Full Stack Bootcamp 👢? Here are some of the reasons that I had, that helped me decide–maybe they can help you too! ⬇️**
> 
> Prior to my bootcamp experience, I had been exposed to coding in Python for almost a decade, through my university and work experience as a Data Analyst. I always enjoyed the building part of the work and wanted to build more 🛠️. I could create a data pipeline or automate a task, but there was no way I could create my own end-to-end app or contribute to a codebase. I tried to do some self-learning but often felt lost with all of the new concepts. After some research and internal back and forth, I decided a bootcamp was right for me.
> 
> If you’re on GitHub, you most likely fall into one of three buckets; (1) someone with programming experience, (2) a GitHub employee, or (3) someone looking to learn more about programming. If you’re in that last bucket, keep reading to hear my experiences with a Full Stack Bootcamp through [Columbia University](https://bootcamp.cvn.columbia.edu/coding/) and get a sense if a bootcamp is right for you 👍.
> 
> ### Why I Chose the Bootcamp Route
> As mentioned, I had previous experience in Python. You can’t get a degree in Data Analysis or work as a true Data Analyst without knowing some Python 🐍, but it’s typically going to be focused on data analysis, engineering, and science use cases such as statistical modeling or data pipelines.
> 
> One day I had an idea for a website (or app) that I thought would be great to try to build. It felt like it would fit a niche need and not be too hard to build. I was wrong. Not knowing where to start, I did some Googling and YouTubing and found some tutorials on Django, and hoped my Python knowledge would provide me with a good enough base. I was immediately lost and didn’t know what anything meant, given I barely knew the basics of HTML. Self-learning was not going to be enough to get this off the ground.
> 
> A few weeks passed and I hadn’t made a dent in the self-learning books I’d purchased and hadn’t written a line of code but I really wanted to, I just felt hopeless. I saw bootcamp ads everywhere for the Flatiron School, Columbia University, and various other options and decided I would apply and take a step in that direction.
> 
> ### How I Chose My Bootcamp
> Choosing a bootcamp wasn’t all that straightforward but it eventually came down to a handful of criteria. I paid close attention to the class schedules, duration of the program, cost 💰, and the syllabus.
> 
> Most important thing for me was the bootcamp cost. If I couldn’t afford it, I wasn’t going to do it. Everyone’s budget is different and this was the best option for me. I chose a program that was the average price I’d seen in my research but also offered flexible payment plans to avoid inundating myself with debt. While the cost might be a lot for many, there are some programs that offer discounts to marginalized groups so make sure to do your own research.
> 
> The bootcamp length and schedule were also key factors for me, as I didn’t want to spend an entire year in school. I wanted to quickly get the skills I needed while being able to maintain my full-time job. I found that part-time was typically twice the duration of full-time programs and the same cost, but with price being a key factor, continuous income was important!
> 
> Finally, I paid close attention to the syllabus for each bootcamp. I took into account the types of projects I wanted to develop and skills necessary to move forward. I cross-referenced the syllabus with those desires and researched the most common skillsets and languages of software developers. Picking a bootcamp that teaches you outdated or niche tools isn’t great, unless you have very specific outcome goals. Look up the role you might want for your next career and find a bootcamp that will put those required skills on your resume ✅.
> 
> ### Bootcamp Experience
> Overall, my bootcamp experience was fast-paced. I had some late nights after work trying to debug code 🐞, or simply just working through an assigned project because I was on a roll and didn’t want to stop. Those moments made me realize this was a good investment for me.
> 
> The bootcamp offered great instructors who were available for help, smart, kind, and diverse students to work with, and lessons on the topics I was looking to learn. I’m currently taking some self-learning courses that feel like a walk in the park because of the solid foundation I received in bootcamp. I’m working through my own app development more confidently this time around.
> 
> If you’ve been wondering how to make your next move into software development, I’d highly recommend taking a bootcamp to accelerate that move!

Ok

**Reply by gpnarks** [reply]
Sounds ok

**Comment by gpnarks** [comment]
It was Awesome

**Reply by Radman0915** [reply]
Ok

**Comment by tphundey** [comment]
good

**Comment by haowang6666** [comment]
> **Thinking about taking a Full Stack Bootcamp 👢? Here are some of the reasons that I had, that helped me decide–maybe they can help you too! ⬇️**
> 
> Prior to my bootcamp experience, I had been exposed to coding in Python for almost a decade, through my university and work experience as a Data Analyst. I always enjoyed the building part of the work and wanted to build more 🛠️. I could create a data pipeline or automate a task, but there was no way I could create my own end-to-end app or contribute to a codebase. I tried to do some self-learning but often felt lost with all of the new concepts. After some research and internal back and forth, I decided a bootcamp was right for me.
> 
> If you’re on GitHub, you most likely fall into one of three buckets; (1) someone with programming experience, (2) a GitHub employee, or (3) someone looking to learn more about programming. If you’re in that last bucket, keep reading to hear my experiences with a Full Stack Bootcamp through [Columbia University](https://bootcamp.cvn.columbia.edu/coding/) and get a sense if a bootcamp is right for you 👍.
> 
> ### Why I Chose the Bootcamp Route
> As mentioned, I had previous experience in Python. You can’t get a degree in Data Analysis or work as a true Data Analyst without knowing some Python 🐍, but it’s typically going to be focused on data analysis, engineering, and science use cases such as statistical modeling or data pipelines.
> 
> One day I had an idea for a website (or app) that I thought would be great to try to build. It felt like it would fit a niche need and not be too hard to build. I was wrong. Not knowing where to start, I did some Googling and YouTubing and found some tutorials on Django, and hoped my Python knowledge would provide me with a good enough base. I was immediately lost and didn’t know what anything meant, given I barely knew the basics of HTML. Self-learning was not going to be enough to get this off the ground.
> 
> A few weeks passed and I hadn’t made a dent in the self-learning books I’d purchased and hadn’t written a line of code but I really wanted to, I just felt hopeless. I saw bootcamp ads everywhere for the Flatiron School, Columbia University, and various other options and decided I would apply and take a step in that direction.
> 
> ### How I Chose My Bootcamp
> Choosing a bootcamp wasn’t all that straightforward but it eventually came down to a handful of criteria. I paid close attention to the class schedules, duration of the program, cost 💰, and the syllabus.
> 
> Most important thing for me was the bootcamp cost. If I couldn’t afford it, I wasn’t going to do it. Everyone’s budget is different and this was the best option for me. I chose a program that was the average price I’d seen in my research but also offered flexible payment plans to avoid inundating myself with debt. While the cost might be a lot for many, there are some programs that offer discounts to marginalized groups so make sure to do your own research.
> 
> The bootcamp length and schedule were also key factors for me, as I didn’t want to spend an entire year in school. I wanted to quickly get the skills I needed while being able to maintain my full-time job. I found that part-time was typically twice the duration of full-time programs and the same cost, but with price being a key factor, continuous income was important!
> 
> Finally, I paid close attention to the syllabus for each bootcamp. I took into account the types of projects I wanted to develop and skills necessary to move forward. I cross-referenced the syllabus with those desires and researched the most common skillsets and languages of software developers. Picking a bootcamp that teaches you outdated or niche tools isn’t great, unless you have very specific outcome goals. Look up the role you might want for your next career and find a bootcamp that will put those required skills on your resume ✅.
> 
> ### Bootcamp Experience
> Overall, my bootcamp experience was fast-paced. I had some late nights after work trying to debug code 🐞, or simply just working through an assigned project because I was on a roll and didn’t want to stop. Those moments made me realize this was a good investment for me.
> 
> The bootcamp offered great instructors who were available for help, smart, kind, and diverse students to work with, and lessons on the topics I was looking to learn. I’m currently taking some self-learning courses that feel like a walk in the park because of the solid foundation I received in bootcamp. I’m working through my own app development more confidently this time around.
> 
> If you’ve been wondering how to make your next move into software development, I’d highly recommend taking a bootcamp to accelerate that move!

**Comment by rjrifatvai7** [comment]
Git clone https://github.com/SIFAT-XD/GF-BF-HACK

cd GF-BF-HACK

python GF-BF.py

**Comment by rjrifatvai7** [comment]
Git clone https://github.com/SIFAT-XD/GF-BF-HACK

cd GF-BF-HACK

python GF-BF.py

**Comment by CelioAmaral** [comment]
Bootcamp is a preparation stage for your development, it is an immersion in knowledge of high relevance for the job market.

Many companies use Bootcamp as a selection process for the job market.

Selecting the best students.. 👍

**Comment by dominikbraun** [comment]
I'm afraid a Bootcamp is one of the most efficient ways to stay unemployed for 6+ months. The entry-level market is oversaturated and the economy is still tough.

**Comment by crazo7924** [comment]
This sounds more like an article and not a question IMHO

**Comment by MuhammadMujtabaa** [comment]
I am keen to be a part of remote bootcamps rather than instead of onsite
I think BootCamp provides an opportunity to explore and learn with speed and momentum
